pkg/kubectl/describe/versioned/describe.go

```diff
@@ -302,8 +302,15 @@ func printUnstructuredContent(w PrefixWriter, level int, content map[string]inte
 }
 
 func smartLabelFor(field string) string {
-	commonAcronyms := []string{"API", "URL", "UID", "OSB", "GUID"}
+	// skip creating smart label if field name contains
+	// special characters other than '-'
+	if strings.IndexFunc(field, func(r rune) bool {
+		return (r < 'A' || r > 'z') && r != '-'
+	}) != -1 {
+		return field
+	}
```

Comment:
Any particular reason to not use regex here instead?

```go
re := regexp.MustCompile(`[0-9\.`]
if re.MatchString(field) {
    return field
}
```

I'm also not sure that we need to exclude `_` from the smart labeling, since it looks like it handles `_` below.

Reply:
I did a comparison of time taken by regex vs strings index func as in the PR and found that the regex was taking about 60% more time compared to the string index function.

Regex used:

**re := regexp.MustCompile(`[~!@#.$%^&*()\[\]0-9/_+=,:;{}|]+`)**

I excluded '-' from the check to skip smart labels. If a field has name like: 'tag-name', a smart label will still be created as: 'Tag - Name'.

Reply:
Your regex is greedy, that probably doesn't help. Aren't you trying to match one of the character? why do you need the + at the end?

Also, I don't understand why we have this feature in the first place ... but sure.

Reply:
I ran the comparison test with the non greedy single line regular expression:

`re := regexp.MustCompile(`(?Us)[~!@#.$%^&*()\[\]0-9/_+=,:;{}|]`)`

Each test runs the match for 1000 iterations. The string index function is still the faster option:

```
> time stringfunc
real	0m0.006s
user	0m0.002s
sys	0m0.003s

> time regexp
real	0m0.010s
user	0m0.006s
sys	0m0.003s
```

The smart label feature tries to make the unstructured content output more user friendly:

With smart label:
```
        Controlplane:
          Id:  sg-0fxxxxxxxxxx
          Ingress Rule:
            Cidr Blocks:
              0.0.0.0/0
            Description:                Kubernetes API
            From Port:                  6443
            Protocol:                   tcp
            Source Security Group Ids:  <nil>
            To Port:                    6443
            Cidr Blocks:                <nil>
            Description:                SSH
```

Without smart label:
```
       controlplane:
          id:  sg-0fxxxxxxxxxx
          ingressRule:
            cidrBlocks:
              0.0.0.0/0
            description:             Kubernetes API
            fromPort:                6443
            protocol:                tcp
            sourceSecurityGroupIds:  <nil>
            toPort:                  6443
            cidrBlocks:              <nil>
            description:             SSH
```